cmake/cmakelibs.cmake

```diff
@@ -0,0 +1,180 @@
+  endif()
+
+  get_property(include_folder_list GLOBAL PROPERTY OSQUERY_EXTENSION_GROUP_INCLUDE_FOLDERS)
+  if(NOT "${include_folder_list}" STREQUAL "")
+    target_include_directories("${OSQUERY_EXTENSION_GROUP_NAME}" PRIVATE
+      ${include_folder_list}
+    )
+  endif()
+endfunction()
+
+function(add_osquery_extension_ex class_name extension_type extension_name ${ARGN})
+  # Make sure the extension type is valid
+  if(NOT "${extension_type}" STREQUAL "config" AND NOT "${extension_type}" STREQUAL "table")
+    message(FATAL_ERROR "Invalid extension type specified")
+  endif()
+
+  # Update the initializer list; this will be added to the main.cpp file of the extension
+  # group
+  set_property(GLOBAL APPEND_STRING
+    PROPERTY OSQUERY_EXTENSION_GROUP_INITIALIZERS
+    "REGISTER_EXTERNAL(${class_name}, \"${extension_type}\", \"${extension_name}\");\n"
+  )
+
+  # Loop through each argument
+  foreach(argument ${ARGN})
+    if("${argument}" STREQUAL "SOURCES" OR "${argument}" STREQUAL "LIBRARIES" OR
+      "${argument}" STREQUAL "INCLUDEDIRS" OR "${argument}" STREQUAL "MAININCLUDES")
+      set(current_scope "${argument}")
+      continue()
+    endif()
+
+    if("${current_scope}" STREQUAL "SOURCES")
+      if(NOT IS_ABSOLUTE "${argument}")
+        set(argument "${CMAKE_CURRENT_SOURCE_DIR}/${argument}")
+      endif()
+      list(APPEND source_file_list "${argument}")
+
+    elseif("${current_scope}" STREQUAL "INCLUDEDIRS")
+      if(NOT IS_ABSOLUTE "${argument}")
+        set(argument "${CMAKE_CURRENT_SOURCE_DIR}/${argument}")
+      endif()
+      list(APPEND include_folder_list "${argument}")
+
+    elseif("${current_scope}" STREQUAL "LIBRARIES")
+      list(APPEND library_list "${argument}")
+
+    elseif("${current_scope}" STREQUAL "MAININCLUDES")
+      list(APPEND main_include_list "${argument}")
+    else()
+      message(FATAL_ERROR "Invalid scope")
+    endif()
+  endforeach()
+
+  # Validate the arguments
+  if("${source_file_list}" STREQUAL "")
+    message(FATAL_ERROR "Source files are missing")
+  endif()
+
+  if("${main_include_list}" STREQUAL "")
+    message(FATAL_ERROR "The main include list is missing")
+  endif()
+
+  # Update the global properties
+  set_property(GLOBAL APPEND PROPERTY OSQUERY_EXTENSION_GROUP_SOURCES
+    ${source_file_list}
+  )
+
+  set_property(GLOBAL APPEND PROPERTY OSQUERY_EXTENSION_GROUP_MAIN_INCLUDES
+    ${main_include_list}
+  )
+
+  if(NOT "${library_list}" STREQUAL "")
+    set_property(GLOBAL APPEND PROPERTY OSQUERY_EXTENSION_GROUP_LIBRARIES
+      ${library_list}
+    )
+  endif()
+
+  if(NOT "${include_folder_list}" STREQUAL "")
+    set_property(GLOBAL APPEND PROPERTY OSQUERY_EXTENSION_GROUP_INCLUDE_FOLDERS
+      ${include_folder_list}
+    )
+  endif()
+endfunction()
+
+macro(ADD_OSQUERY_EXTENSION TARGET)
+  add_executable(${TARGET} ${ARGN})
+  set_target_properties(${TARGET} PROPERTIES OUTPUT_NAME "${TARGET}.ext")
+endmacro(ADD_OSQUERY_EXTENSION)
+
+macro(ADD_OSQUERY_MODULE TARGET)
```

Comment:
This should be converted into a function

The naming scheme used in the project follows camelCase
https://github.com/osquery/osquery/blob/52171f5676f64c59735cb960ef5fd52601dccdbc/cmake/utilities.cmake#L8

Variables, unless they end up in the global scope, are named lowercase